packages/cashier/src/Containers/crypto-deposit.jsx

```diff
@@ -31,32 +31,68 @@ const CryptoDeposit = ({
         pollApiForDepositAddress(false);
     }, [pollApiForDepositAddress]);
 
-    const eth_option_list = [
+    const option_list = [
         { text: <Localize i18n_default_text='Binance Smart Chain' />, value: 1 },
-        { text: <Localize i18n_default_text='Ethereum (ERC20)' />, value: 2 },
-        { text: <Localize i18n_default_text='Ethereum (ETH)' />, value: 3 },
+        { text: <Localize i18n_default_text='Polygon (Matic)' />, value: 2 },
+        { text: <Localize i18n_default_text='Tron' />, value: 3 },
+        { text: <Localize i18n_default_text='Ethereum (ERC20)' />, value: 4 },
+        { text: <Localize i18n_default_text='Ethereum (ETH)' />, value: 5 },
     ];
 
-    const [eth_option_message, setEthOptionMessage] = useState('');
-    const [eth_option_list_value, setEthOptionListValue] = useState(0);
-    const [eth_qrcode_header, setEthQRCodeHeader] = useState('');
+    const [option_message, setOptionMessage] = useState('');
+    const [option_list_value, setOptionListValue] = useState(0);
+    const [qrcode_header, setQRCodeHeader] = useState('');
 
-    const onChangeEthListOption = event => {
-        if (event.target.value === eth_option_list[0].value) {
-            setEthOptionMessage(
-                <Localize i18n_default_text='We do not support Binance Smart Chain tokens to deposit, please use only Ethereum (ETH).' />
-            );
-        } else if (event.target.value === eth_option_list[1].value) {
-            setEthOptionMessage(
-                <Localize i18n_default_text='This is an Ethereum (ETH) only address, please do not use ERC20 tokens.' />
+    const onChangeListOption = event => {
+        const token_ETH = 'ETH';
+        const token_USDC_eUSDT = 'ERC20';
+
+        const token = currency === 'ETH' ? token_ETH : ['USDC', 'eUSDT'].includes(currency) ? token_USDC_eUSDT : '';
+
+        const setProhibitedTokenMessage = () => {
+            const prohibited_token = token === token_ETH ? `${token_USDC_eUSDT} token` : token_ETH;
+            setOptionMessage(
+                <Localize
+                    i18n_default_text='This is an Ethereum ({{token}}) only address, please do not use {{prohibited_token}}.'
+                    values={{ token, prohibited_token }}
+                />
             );
-        } else if (event.target.value === eth_option_list[2].value) {
-            setEthOptionMessage('');
-            setEthQRCodeHeader(
+        };
+
+        const setQRCodeHeaderMessage = () => {
+            setOptionMessage('');
+            setQRCodeHeader(
                 <Localize i18n_default_text="Do not send any other currency to the following address. Otherwise, you'll lose funds." />
             );
+        };
+
+        if (event.target.value === option_list[0].value) {
```

Comment:
@george-usynin-deriv Isn't it nicer to use `switch` instead of this long and nested `if` statement?